scripts/make-bundle

```diff
@@ -12,171 +12,196 @@ interface for those not familiar with the python
-    # Then we need to create a dir where all the packages
-    # will come from.
-    os.mkdir(os.path.join(dirname, 'packages'))
-    return dirname
-
-
-def download_package_tarballs(dirname, packages):
-    with cd(dirname):
-        for package in packages:
-            run('pip install -d . %s==%s %s' % (
-                package, PACKAGE_VERSION[package], INSTALL_ARGS))
-
-
-def download_cli_deps(scratch_dir):
-    # As far as i can tell this is a bug in pip.
-    # Running 'pip intall -d . ~/clidir' does not work.
-    # Running 'pip install -d foo ~/clidir' results
-    # in python consuming 100% CPU.
-    # This problem is surprisingly hard, specifically that
-    # of downloading all the dependent python packages of
-    # the awscli as tarballs.
-    # So here's what we do.
-    # Step one create a virtualenv.
-    venv_dir = tempfile.mkdtemp()
-    run('virtualenv %s' % venv_dir)
-    pip = os.path.join(venv_dir, 'bin', 'pip')
-    assert os.path.isfile(pip)
-    awscli_dir = os.path.dirname(
-        os.path.dirname(os.path.abspath(__file__)))
-    with cd(scratch_dir):
-        run('%s install %s -d . -e %s' % (pip, INSTALL_ARGS, awscli_dir))
-    # Remove the awscli package, we'll create our own sdist.
-    _remove_cli_zip(scratch_dir)
-
-
-def _remove_cli_zip(scratch_dir):
-    clidir = [f for f in os.listdir(scratch_dir) if f.startswith('awscli')]
-    assert len(clidir) == 1
-    os.remove(os.path.join(scratch_dir, clidir[0]))
-
-
-def add_cli_sdist(scratch_dir):
-    awscli_dir = os.path.dirname(
-        os.path.dirname(os.path.abspath(__file__)))
-    if os.path.exists(os.path.join(awscli_dir, 'dist')):
-        shutil.rmtree(os.path.join(awscli_dir, 'dist'))
-    with cd(awscli_dir):
-        run('python setup.py sdist')
-        filename = os.listdir('dist')[0]
-        shutil.move(os.path.join('dist', filename),
-                    os.path.join(scratch_dir, filename))
-
-
-def create_bootstrap_script(scratch_dir):
+
+
+PYTHON_VERSIONS = ["2.6", "2.7", "3.3", "3.4", "3.5", "3.6"]
+
+
+def pip(python_version):
+    if python_version == "2.6":
+        return "pip.__main__"
+    return "pip"
+
+
+def ensure_packages(requirements, *, output_dir, pythons=None):
+    if pythons is None:
+        pythons = PYTHON_VERSIONS
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # We want to loop over every version of Python we support and download
+        # or build wheels for them.
+        first = False
+        for python in sorted(pythons, reverse=True):
+            print("Downloading Dependencies for {}".format(python))
+
+            # Create our virtual environment.
+            venv_path = os.path.abspath(os.path.join(tmpdir, python))
+            python_path = os.path.join(venv_path, "bin", "python")
+            subprocess.run(
+                ["virtualenv", "-ppython{}".format(python), venv_path],
+                stdout=subprocess.PIPE,
+                check=True,
```

Comment:
So I was doing ``subprocess.PIPE`` largely to hide the output, but I didn't want to hide the ``stderr`` output so that it didn't silence warnings/errors. So I think that unless we're managing the ``stdout``/``stderr`` output in a way that is more complex than either not outputting it or outputting it then we don't want to set that here.